### ISSUE TITLE
[Performance] OpenVINO crashes without logs

### ISSUE THREAD

**Issue by SimonRelu** [microsoft/onnxruntime#16662]
### Describe the issue

When I run using openVINO the program just exits without error. I'm using
OS: Windows 11
CPU: 11th Gen Intel Core i5-11400H

### To reproduce

```
import onnxruntime
print("loading model")
session = onnxruntime.InferenceSession(
    "vino.onnx",
    providers=["OpenVINOExecutionProvider"],
    provider_options=[{"device_type": "GPU.0_FP16"}],
)
session = onnxruntime.InferenceSession(
    "vino.onnx",
    providers=["OpenVINOExecutionProvider"],
    provider_options=[{"device_type": "CPU_FP16"}],
)
options = onnxruntime.RunOptions()
options.log_severity_level = 0
x = np.random.randn(1, 1, 96, 96, 96).astype(np.float16)
print(session.run(None, {session.get_inputs()[0].name: x}, run_options=options))
```
I get the following logs:
```
loading model
starting
2023-07-11 16:58:58.1133776 [I:onnxruntime:, sequential_executor.cc:514 onnxruntime::ExecuteThePlan] Number of streams: 1
```
and when using `GPU.0_FP16` it gets stuck during loading model:
```
loading model
```

Any idea how I could debug this? I've tried this on 2 sepperate PCs and got the exact same issue.

### Urgency

blocking

### Platform

Windows

### OS Version

11

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

1.14.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

OpenVINO

### Execution Provider Library Version

2023.0.1

### Is this a quantized model?

No

**Comment by jywu-msft** [comment]
@sfatimar , @saurabhintel0  any suggestions?

**Comment by jywu-msft** [comment]
adding @sspintel

**Comment by saurabhintel0** [comment]
@SimonRelu 
Can you please share the model if possible.
Was not able to reproduce the issue with the sample code.

If you want to enable logs set the environment variable
set ORT_OPENVINO_ENABLE_CI_LOG=1
set ORT_OPENVINO_ENABLE_DEBUG=1

**Comment by SimonRelu** [comment]
Sorry for the late response. So it seems that the original issue was related to a mistake in our onnx conversion. However, now that that is fixed I get the following error:
```
In the OpenVINO EP
2023-07-20 12:50:42.091734281 [W:onnxruntime:, session_state.cc:1136 VerifyEachNodeIsAssignedToAnEp] Some nodes were not assigned to the preferred execution providers which may or may not have an negative impact on performance. e.g. ORT explicitly assigns shape related ops to CPU to improve perf.
2023-07-20 12:50:42.091751691 [W:onnxruntime:, session_state.cc:1138 VerifyEachNodeIsAssignedToAnEp] Rerunning with verbose output on a non-minimal build will show node assignments.
starting
CreateNgraphFunc
<kernel>:915:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[0] * weight[0];
        ^     ~
<kernel>:915:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[0] * weight[0];
                   ^      ~
<kernel>:916:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[1] * weight[1];
        ^     ~
<kernel>:916:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[1] * weight[1];
                   ^      ~
<kernel>:917:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[2] * weight[2];
        ^     ~
<kernel>:917:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[2] * weight[2];
                   ^      ~
<kernel>:918:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[3] * weight[3];
        ^     ~
<kernel>:918:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[3] * weight[3];
                   ^      ~
<kernel>:921:71: warning: GCC does not allow overloadable attribute in this position on a function definition
....
<kernel>:922:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[0] * weight[0];
        ^     ~
<kernel>:922:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[0] * weight[0];
                   ^      ~
<kernel>:923:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[1] * weight[1];
        ^     ~
<kernel>:923:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[1] * weight[1];
                   ^      ~
<kernel>:924:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[2] * weight[2];
        ^     ~
<kernel>:924:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[2] * weight[2];
                   ^      ~
<kernel>:925:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[3] * weight[3];
        ^     ~
<kernel>:925:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[3] * weight[3];
                   ^      ~
<kernel>:928:72: warning: GCC does not allow overloadable attribute in this position on a function definition
....
<kernel>:929:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[0] * weight[0];
        ^     ~
<kernel>:929:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[0] * weight[0];
                   ^      ~
<kernel>:930:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[1] * weight[1];
        ^     ~
<kernel>:930:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[1] * weight[1];
                   ^      ~
<kernel>:931:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[2] * weight[2];
        ^     ~
<kernel>:931:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[2] * weight[2];
                   ^      ~
<kernel>:932:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[3] * weight[3];
        ^     ~
<kernel>:932:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[3] * weight[3];
                   ^      ~
<kernel>:935:73: warning: GCC does not allow overloadable attribute in this position on a function definition
....
<kernel>:936:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[0] * weight[0];
        ^     ~
<kernel>:936:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[0] * weight[0];
                   ^      ~
<kernel>:937:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[1] * weight[1];
        ^     ~
<kernel>:937:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[1] * weight[1];
                   ^      ~
<kernel>:938:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[2] * weight[2];
        ^     ~
<kernel>:938:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[2] * weight[2];
                   ^      ~
<kernel>:939:9: error: subscripted access is not allowed for OpenCL vectors
 acc += input[3] * weight[3];
        ^     ~
<kernel>:939:20: error: subscripted access is not allowed for OpenCL vectors
 acc += input[3] * weight[3];
                   ^      ~
<kernel>:956:1: warning: GCC does not allow overloadable attribute in this position on a function definition
....
Traceback (most recent call last):
  File "scripts/Simon/onnx/vino.py", line 33, in <module>
    print(session.run(None, {session.get_inputs()[0].name: x}))
  File "/home/hector/simon/1-2-ai-monorepo/env/lib/python3.8/site-packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 200, in run
    return self._sess.run(output_names, input_feed, run_options)
onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Encountered unknown exception in Run()
```

I did verify that this model runs correctly on cpu and cuda. This happens when I set device to `GPU_FP16`. on `CPU_FP16` it runs fine.

**Comment by menggui1993** [comment]
@SimonRelu Hi, Simon. I met the same opencl error. Did you find any solutions?

**Comment by sspintel** [comment]
Hi @SimonRelu, @menggui1993,

The error might be due to outdated graphics drivers / OpenCL. Subscripted accesses of arrays are legal and widely used, so I suspect this could be a driver issue. Consider updating to [latest GPU drivers](https://www.intel.com/content/www/us/en/search.html#sort=relevancy&f:@tabfilter=[Downloads]&f:@stm_10385_en=[Graphics]) and updating your OpenCL

**Comment by SimonRelu** [comment]
Hi, I believe for me the issue was resolved by going to OpenVINO version 2023.0.1 from 2023.0.0 but it could also be that I updated the version inbetween. I'll close this PR for now since the issue has been resolved for me
